app/plugins/Waivers/src/Model/Table/GatheringWaiverExemptionsTable.php

```
@@ -0,0 +1,141 @@
+     * @param \Cake\ORM\RulesChecker $rules The rules object to be modified.
+     * @return \Cake\ORM\RulesChecker
+     */
+    public function buildRules(RulesChecker $rules): RulesChecker
+    {
+        $rules->add($rules->existsIn(['gathering_activity_id'], 'GatheringActivities'), ['errorField' => 'gathering_activity_id']);
+        $rules->add($rules->existsIn(['waiver_type_id'], 'WaiverTypes'), ['errorField' => 'waiver_type_id']);
+        $rules->add($rules->existsIn(['member_id'], 'Members'), ['errorField' => 'member_id']);
+
+        // Ensure only one exemption per activity/waiver type combination
+        $rules->add(
+            function ($entity, $options) {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$options'. (undefined)

(UnusedFormalParameter)